lib/widgets/compose_box.dart

```diff
@@ -234,61 +307,110 @@ class _AttachFileButton extends StatelessWidget {
       return;
     }
 
-    final store = PerAccountStoreWidget.of(context);
+    final Iterable<_File> files = result.files.map((f) {
+      assert(f.readStream != null);  // We passed `withReadStream: true` to pickFiles.
+      return _File(content: f.readStream!, length: f.size, filename: f.name);
+    });
+    _uploadFiles(context: context, contentController: contentController, contentFocusNode: contentFocusNode,
+      files: files);
+  }
 
-    final List<PlatformFile> tooLargeFiles = [];
-    final List<PlatformFile> rightSizeFiles = [];
-    for (PlatformFile file in result.files) {
-      if ((file.size / (1 << 20)) > store.maxFileUploadSizeMib) {
-        tooLargeFiles.add(file);
-      } else {
-        rightSizeFiles.add(file);
-      }
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(icon: const Icon(Icons.attach_file), onPressed: () => _handlePress(context));
+  }
+}
+
+class _AttachMediaButton extends StatelessWidget {
+  const _AttachMediaButton({required this.contentController, required this.contentFocusNode});
+
+  final ContentTextEditingController contentController;
+  final FocusNode contentFocusNode;
+
+  _handlePress(BuildContext context) async {
```

Comment:
Similarly
```suggestion
  dynamic _handlePress(BuildContext context) async {
```

and I guess I missed that previously on `_AttachFileButton`.

Reply:
Hmm, I don't understand this one yet, and what are you referring to with "similarly"? Do you mean `Future<void>`, or should it really be `dynamic`?

Reply:
By "similarly" I mean to not have an implicit `dynamic`.

But probably we don't actually want `dynamic`, and `void` is better.